test/integration/http_timeout_integration_test.cc

```diff
@@ -499,4 +501,59 @@ void HttpTimeoutIntegrationTest::testRouterRequestAndResponseWithHedgedPerTryTim
   EXPECT_EQ("200", response->headers().getStatusValue());
 }
 
+// Starts a request with a header timeout specified, sleeps for longer than the
+// timeout, and ensures that a timeout is received.
+TEST_P(HttpTimeoutIntegrationTest, RequestHeaderTimeout) {
+  config_helper_.addConfigModifier(
+      [&](envoy::extensions::filters::network::http_connection_manager::v3::HttpConnectionManager&
+              hcm) {
+        auto* request_headers_timeout = hcm.mutable_request_headers_timeout();
+        request_headers_timeout->set_seconds(1);
+        request_headers_timeout->set_nanos(0);
+      });
+  setDownstreamProtocol(Http::CodecClient::Type::HTTP1);
```

Comment:
I'd be inclined to return if the protocol isn't HTTP/1, so we don't run this twice (and the H2 test confusingly runs HTTP/1)